include/core/memory.hpp

```diff
@@ -110,29 +111,65 @@ class Memory {
    * \param amount Number of Bytes comprising the value
    * \returns MemoryValue holding the data stored in the Memory at
    *          [address;address+amount[
+   * \throws AssertionError iff [address; address+amount[ is not subset of
+   *         [0; _byteCount[
    */
-  MemoryValue
-  get(const std::size_t address, const std::size_t amount = 1) const;
+  MemoryValue get(std::size_t address, std::size_t amount = 1) const;
+  /**
+   * \brief Returns a MemoryValue holding the data stored in the Memory at
+   *        [address; address+amount[
+   * \param address Starting location of the Value in the Memory
+   * \param amount Number of Bytes comprising the value
+   * \returns MemoryValue holding the data stored in the Memory at
+   *          [address;address+amount[ with 0 for all bytes not in
+   *          [0; _byteCount[
+   */
+  MemoryValue tryGet(std::size_t address, std::size_t amount = 1) const;
   /**
    * \brief Writes value into the Memory at address
    * \param address Starting address of the to be overwritten value
    * \param value Value to write
    * \param ignoreProtection if this is true do ignore all protection
+   * \throws AssertionError iff [address; address+(value.size()/_byteSize)[ is
+   *         not subset of [0; _byteCount[ or value.size()%_byteSize != 0
    */
-  void put(const std::size_t address,
+  void put(std::size_t address,
            const MemoryValue &value,
            bool ignoreProtection = false);
   /**
+   * \brief Writes value into the Memory at address
+   * \param address Starting address of the to be overwritten value
+   * \param value Value to write
+   * \param ignoreProtection if this is true do ignore all protection
+   */
+  void tryPut(std::size_t address,
+              const MemoryValue &value,
+              bool ignoreProtection = false);
+  /**
    * \brief Writes value into the Memory at address and returns the previous
    *        value
    * \param address Starting address of the to be overwritten value
    * \param value Value to write
    * \param ignoreProtection if this is true do ignore all protection
    * \returns Value that was overwritten
+   * \throws AssertionError iff [address; address+(value.size()/_byteSize)[ is
+   *         not subset of [0; _byteCount[ or value.size()%_byteSize != 0
    */
-  MemoryValue set(const std::size_t address,
+  MemoryValue set(std::size_t address,
```

Comment:
If you use `using size_t = std::size_t` at the top of the class, you can just use `size_t`